samples/net/sockets/echo_async/src/socket_echo.c

```diff
@@ -102,19 +102,23 @@ int main(void)
 {
 	int res;
 	static int counter;
-	int serv4, serv6;
+	int num_servs = 0;
+	int serv4;
 	struct sockaddr_in bind_addr4 = {
 		.sin_family = AF_INET,
 		.sin_port = htons(PORT),
 		.sin_addr = {
 			.s_addr = htonl(INADDR_ANY),
 		},
 	};
+#ifdef CONFIG_NET_IPV6
```

Comment:
The sample can be used both in Zephyr and Linux, but Linux does not know anything about CONFIG_NET_IPV6. So this needs to be changed here.

Reply:
Yeah, in a sense that adding this #ifdef would blindly disable it for Linux (or another POSIX OS). I'd suggest to use something like `#ifdef USE_IPV6`, and set USE_IPV6 in the header of this file, where Zephyr-vs-POSIX ifdef already is.

Reply:
Thanks, done.